algorithms/estonian/stemmer.tt

```diff
@@ -0,0 +1,479 @@
+[% header('Estonian stemming algorithm') %]
+
+<h2>Links to resources</h2>
+
+<ul>
+[% algorithm_lis('estonian', 'Estonian') %]
+</ul>
+
+<div style="background-color:lightpink">
+
+<p>
+Here is a sample of vocabulary, with the stemmed forms that will
+be generated with the algorithm.
+</p>
+
+[% algorithm_vocab('estonian', [
+'raamat',
+'raamatu',
+'raamatut',
+'raamatule',
+'raamatud',
+'raamatute',
+'raamatuid',
+'raamatutele',
+'raamatutestki',
+'hele',
+'heleda',
+'heledat',
+'heledale',
+'heledad',
+'heledate',
+'heledaid',
+'heledatele',
+'heledam',
+'heledama',
+'heledamat',
+'heledamad',
+'heledamate',
+'heledamaid',
+'heledamatelegi',
+'heledaim',
+'heledaima',
+'heledaimat',
+'heledaimale',
+'heledaimad',
+'heledaimate',
+'heledaimaid',
+'heledaimatelt',
+'hobune',
+'hobuse',
+'hobust',
+'hobusele',
+'hobused',
+'hobuste',
+'hobuseid',
+'hobustele',
+], [
+'hüpata',
+'hüppa',
+'hüppaksin',
+'hüppaksid',
+'hüppaks',
+'hüppaksime',
+'hüppaksite',
+'hüppan',
+'hüppad',
+'hüppab',
+'hüppame',
+'hüppate',
+'hüppavad',
+'hüppasin',
+'hüppasid',
+'hüppas',
+'hüppasime',
+'hüppasite',
+'hüpanuksite',
+'hüpatakse',
+'hüpati',
+'hüpanud',
+'hüpanutest',
+'hüpates',
+'hüppavat',
+'hüppavatele',
+'hüppamata',
+'hüppamast',
+'hüljes',
+'hülge',
+'hüljest',
+'hülgesse',
+'hüljeste',
+'hülgeid',
+'hüljestesse',
+'hülgeisse',
+'ohutule',
+'ohutud',
+'ohutuid',
+'ohututele',
+]) %]
+</div>
+
+<p>
+This algorithm is written in collaboration with Estonian text analytics enterprise Texta.
+</p>
+
+<p>
+Letters in Estonian include the following accented forms,
+</p>
+
+<DL><DD>
+    <B><I>ä  &nbsp;  ö  &nbsp;  õ  &nbsp;  ü</I></B>
+</DL>
+
+<p>
+The following letters are vowels (<I>V1</I>):
+</p>
+
+<DL><DD>
+    <B><I>a  &nbsp;  e  &nbsp;  i  &nbsp;  o  &nbsp;  u  &nbsp;  õ  &nbsp;  ä  &nbsp;  ö  &nbsp;  ü</I></B>
+</DL>
+
+<p>
+<I>RV</I> is defined as one of the following:
+</p>
+
+<DL><DD>
+    <B><I>a  &nbsp;  e  &nbsp;  i  &nbsp;  u  &nbsp;  o</I></B>
+</DL>
+
+<p>
+<I>KI</I> is defined as one of the following (letters possible before <B><I>-ki</I></B> emphasis):
+</p>
+
+<DL><DD>
+    <B><I>k  &nbsp;  p  &nbsp;  t  &nbsp;  g  &nbsp;  b  &nbsp;  d  &nbsp;  s  &nbsp;  h  &nbsp;  f  &nbsp;  š  &nbsp;  z  &nbsp;  ž  &nbsp;  o</I></B>
```

Comment:
This list contains `o` but `KI` in the Snowball source doesn't:

```
define KI 'kptgbdshfšzž'
```

Reply:
Resolved by removing `o` here (it's not in the draft version of this description in `Estonian Snowball 1.3.odt` either, so it seems it was introduced during reformatting).